thirdeye-server/src/main/java/ai/startree/thirdeye/core/AppAnalyticsService.java

```diff
@@ -155,11 +175,44 @@ public AnomalyStatsApi computeAnomalyStats(final DaoFilter filter) {
         ? anomalyFeedbacksSupplier.get()
         : getAnomalyFeedbacks(filter);
     return new AnomalyStatsApi()
-        .setTotalCount(anomalyManager.countParentAnomalies(filter))
-        .setCountWithFeedback((long) allFeedbacks.size())
+        .setTotalCount(countTotal(filter))
+        .setCountWithFeedback(countFeedbacks(filter))
         .setFeedbackStats(aggregateFeedbackTypes(allFeedbacks));
   }
 
+  private Long countTotal(final DaoFilter filter) {
+    final DaoFilter finalFilter = new DaoFilter()
+        .andPredicate(nonChild())
+        .andPredicate(notIgnored());
+    Optional.ofNullable(filter)
+        .ifPresent(daoFilter -> finalFilter.andPredicate(filter.getPredicate()));
+    return anomalyManager.countParentAnomalies(finalFilter);
+  }
+
+  private Long countFeedbacks(final DaoFilter filter) {
+    final DaoFilter finalFilter = new DaoFilter()
+        .andPredicate(hasFeedback());
+    Optional.ofNullable(filter)
+        .ifPresent(daoFilter -> finalFilter.andPredicate(filter.getPredicate()));
+    return countTotal(finalFilter);
+  }
+
+  private Predicate hasFeedback() {
```

Comment:
Oh I didn't made them static as it being a singleton service and the methods are private, no other reason.
The motivation to introduce these methods here was only because the `notIgnored()` predicate was getting used at like 3 places so I pulled the predicate creation out, and then it just looked clean and easy to use for me hence replicated it for others too. 
Handling `DaoFilter` in cases where filters are accumulated as we go down the flow was looking hard to me as it involved get predicate, combine the predicates, set predicate. And every place I needed to check if the predicate from filter is not null and handle it accordingly. Let me know if there is a better way to handle this efficiently.